internal/consumer/message.go

```diff
@@ -42,22 +42,38 @@ type (
 	}
 )
 
+func newDLQMetadata() *DLQMetadata {
+	return &DLQMetadata{
+		RetryCount:  0,
+		Topic:       "",
+		Partition:   -1,
+		Offset:      -1,
+		TimestampNs: -1,
+		Data:        nil,
+	}
+}
+
 // newMessage builds a new Message object from the given kafka message
-func newMessage(scm *sarama.ConsumerMessage, ackID ackID, ackMgr *ackManager, dlq DLQ) *Message {
+func newMessage(scm *sarama.ConsumerMessage, ackID ackID, ackMgr *ackManager, dlq DLQ, metadata DLQMetadata) *Message {
 	return &Message{
 		msg: scm,
 		ctx: msgContext{
 			ackID:  ackID,
 			ackMgr: ackMgr,
 			dlq:    dlq,
 		},
+		metadata: metadata,
 	}
 }
 
 // Key is a mutable reference to the message's key
 func (m *Message) Key() []byte {
 	result := make([]byte, len(m.msg.Key))
```

Comment:
nit: the make here isn't accurate

Reply:
I dont follow. What do you mean?

Reply:
I meant the size of byte array doesn't necessarily have to len(m.msg.Key)

Reply:
Good catch. Fixed.